util/prefix-writer.go

```diff
@@ -0,0 +1,37 @@
+package util
+
+import (
+	"bytes"
+	"io"
+)
+
+func PrefixedWriter(writer io.Writer, prefix string) io.Writer {
+	return &prefixedWriter{writer: writer, prefix: prefix, beginningOfANewLine: true}
+}
+
+func (pf *prefixedWriter) Write(p []byte) (int, error) {
+	buf := bytes.Buffer{}
+
+	for _, b := range p {
+		if pf.beginningOfANewLine {
+			buf.WriteString(pf.prefix)
+			pf.beginningOfANewLine = false
+		}
+
+		buf.WriteByte(b)
```

Comment:
I was wondering why output should be buffered and not directly written to `pf.writer`?

Reply:
I did not want to call the underlying `pf.writer` for each character as it may degrade performance.

This buffering does not cross single `.Write()` call span so it will not create a "typical" buffered i/o effect.

Still, it needs an additional memory and one copying of the input `[]byte`. If you feel there is more efficient way I would be more than happy to apply it.